pkg/rtc/participant.go

```diff
@@ -2032,6 +2057,8 @@ func (p *ParticipantImpl) onAnyTransportNegotiationFailed() {
 func (p *ParticipantImpl) EnqueueSubscribeTrack(trackID livekit.TrackID, f func(sub types.LocalParticipant) error) {
 	p.params.Logger.Infow("queuing subscribe", "trackID", trackID)
 
+	p.supervisor.UpdateSubscription(trackID, true)
```

Comment:
should this happen before while queued? or after it's dequeued?

Reply:
It should happen before as this is the expected operation, i. e. at some point in time there should be a corresponding `subscribedTrack`. So, it is added when there is a subscription request.

Reply:
that makes sense, capturing client intent as well as we can